crates/ruff_python_formatter/src/trivia.rs

```diff
@@ -386,7 +436,14 @@ impl<'a> SimpleTokenizer<'a> {
                 } else if c == '\\' {
                     TokenKind::Continuation
                 } else {
-                    let kind = TokenKind::from_non_trivia_char(c);
+                    let kind = if is_identifier_start(c) {
+                        self.cursor.eat_back_while(is_identifier_continuation);
```

Comment:
We need to test if the last processed character was an identifier start (otherwise 555 is a valid identifier)

Reply:
unless i'm missing something, wouldn't this mean that say `555` would be parsed as a single `Other` backwards, but `[Other, Bogus, Bogus]` forwards?  to fix, don't we need to scan first (without bumping) to make sure we end on an `identifier_start`?

Reply:
Oh yeah. Good point. I think we can either scan first (similar to detecting if there are any comments), or we clone curser, lex it, and restore the cursor if first turns out not to be an identifier 

Did these changes fix the graphemes issue?